src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationContext.cs

```diff
@@ -286,6 +295,14 @@ private DurableHttpRequest CreateHttpRequestMessageCopy(DurableHttpRequest durab
                 content: durableHttpRequest.Content,
                 tokenSource: durableHttpRequest.TokenSource);
 
+            // Do not copy over the x-functions-key header, as in many cases, the
+            // functions key used for the initial request will be a Function-level key
+            // and the status endpoint requires a master key.
+            if (newDurableHttpRequest.Headers.ContainsKey("x-functions-key"))
```

Comment:
Do we need to consider the `code` query string parameter as well?

Reply:
We don't copy over query string parameters, unlike headers.

Reply:
I think it is debatable we should even send any headers from the initial request to the polling request, but that is a breaking change at this point.

Reply:
Agreed. If you think we should remove it in a future version, then I suggest we open an issue to track doing so.

Reply:
#1225